homeassistant/components/apple_tv/media_player.py

```diff
@@ -135,6 +143,18 @@ def async_device_connected(self, atv):
         # Listen to power updates
         self.atv.power.listener = self
 
+        if self.atv.features.in_state(FeatureState.Available, FeatureName.AppList):
+            self.hass.create_task(self._update_app_list())
+
+    async def _update_app_list(self):
+        _LOGGER.debug("Updating app list")
+        try:
+            apps = await self.atv.apps.app_list()
+            self._app_list = {app.name: app.identifier for app in apps}
+            self.async_write_ha_state()
```

Comment:
Please move code that isn't expected to fail out of the try block

Reply:
Fixed